prog/victoria_metrics/victoria_metrics_resource_nexus.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::VictoriaMetrics::VictoriaMetricsResourceNexus < Prog::Base
+  subject_is :victoria_metrics_resource
+
+  def self.assemble(project_id, name, location_id, admin_user, vm_size, storage_size_gib)
+    project = Project[project_id]
+    fail "No existing project" unless project
+
+    location = Location[location_id]
+    fail "No existing location" unless location
+
+    Validation.validate_name(name)
+    Validation.validate_victoria_metrics_username(admin_user)
+    Validation.validate_vm_size(vm_size, "x64")
+    storage_size_gib = Validation.validate_victoria_metrics_storage_size(storage_size_gib)
+
+    DB.transaction do
+      ubid = VictoriaMetricsResource.generate_ubid
+      root_cert_1, root_cert_key_1 = Util.create_root_certificate(common_name: "#{ubid} Root Certificate Authority", duration: 60 * 60 * 24 * 365 * 5)
+      root_cert_2, root_cert_key_2 = Util.create_root_certificate(common_name: "#{ubid} Root Certificate Authority", duration: 60 * 60 * 24 * 365 * 5)
+
+      victoria_metrics_resource = VictoriaMetricsResource.create(
+        name:,
+        location_id: location.id,
+        admin_user:,
+        admin_password: SecureRandom.urlsafe_base64(15),
+        root_cert_1:,
+        root_cert_key_1:,
+        root_cert_2:,
+        root_cert_key_2:,
+        target_vm_size: vm_size,
+        target_storage_size_gib: storage_size_gib,
+        project_id: project.id
+      )
+
+      firewall = Firewall.create_with_id(name: "#{victoria_metrics_resource.ubid}-firewall", location_id: location.id, description: "VictoriaMetrics default firewall", project_id: Config.victoria_metrics_service_project_id)
+
+      private_subnet_id = Prog::Vnet::SubnetNexus.assemble(Config.victoria_metrics_service_project_id, name: "#{victoria_metrics_resource.ubid}-subnet", location_id: location.id, firewall_id: firewall.id).id
+      victoria_metrics_resource.update(private_subnet_id: private_subnet_id)
+      victoria_metrics_resource.set_firewall_rules
+
+      Prog::VictoriaMetrics::VictoriaMetricsServerNexus.assemble(
+        victoria_metrics_resource.id,
+        "#{name}-node-01"
```

Comment:
I agree with @enescakir on this name being not required. I guess this can be useful when we have multiple servers, but even then it is tricky to maintain. Two reasons:
- Over time, we would recycle some servers, so there would be gaps in the numbers.
- Every time we add a new server, you would need to parse existing VM names and find highest used number to generate correct suffix.